src/Dialect/ONNX/ONNXOps/OpHelper.cpp

```diff
@@ -317,6 +317,24 @@ ONNXConstantOp getONNXConstantOp(Value value) {
   return dyn_cast_or_null<ONNXConstantOp>(value.getDefiningOp());
 }
 
+bool getI64ValuesFromSmallONNXConstantOp(
+    mlir::Value val, mlir::SmallVectorImpl<int64_t> &iRes) {
+  ElementsAttr elemsAttr = getElementAttributeFromONNXValue(val);
+  if (!elemsAttr)
+    return false;
+  if (!getElementType(elemsAttr.getType()).isInteger(64))
+    return false;
+
+  DenseElementsAttr denseAttr =
+      ElementsAttrBuilder::toDenseElementsAttr(elemsAttr);
+  auto valueIt = denseAttr.getValues<IntegerAttr>().begin();
+  for (unsigned int i = 0; i < denseAttr.getNumElements(); ++i) {
+    int64_t d = (*valueIt++).cast<IntegerAttr>().getInt();
+    iRes.emplace_back(d);
+  }
```

Comment:
`getValues<T>()` works on the ElementsAttr interface and is supported by DisposableElementsAttr, so there is no need to convert to DenseElementsAttr

we generally access ElementsAttr like this all over the onnx-mlir code base and is why all the code paths for the ONNX dialect work regardless of whether an ElementsAttr is dense or disposable

moreover, it's simpler and faster to access the elements as int64_t, see the example in ConstPropOfShape: https://github.com/onnx/onnx-mlir/blob/main/src/Transform/ONNX/ConstProp.cpp#L905

Reply:
This is what I was looking for. Changed to use `getValues<T>()`. Thanks!